### PR TITLE
test spark when OpenLineage fails

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Signed-off-by: Pawel Leszczynski <leszczynski.pawel@gmail.com>

### Problem

We need to make sure that any OpenLineage related issues won't affect Spark job run. 

Closes: #539

### Solution

Implement integration tests that break OpenLineage event generation and emitting. Verify that Spark job runs successfully at the same time. 

Three integration tests introduced: 
 * incorrect openlineage url 
 * mockserver returns 400 response code
 * mockserver returns 500 response code

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [x] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [x] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    SparkContainerUtils.makePysparkContainerWithDefaultConf(
+            network,
+            "http://openlineageclient:" + CORRECT_PORT,
+            ".*Spark is fine!.*",
+            openLineageClientMockContainer,
+            "testFailures",
+            "/opt/spark_scripts/spark_test_failures.py")
+        .start();
+  }
```
Comment (collado-mike) [inline integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java:88]: These tests are great!

File: integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        spark.createDataFrame(Arrays.asList(new GenericRow(new Object[] {"one"})), schema);
+
+    dataset.write().mode(SaveMode.Overwrite).saveAsTable(FAILING_TABLE_NAME_FAIL_ON_IS_DEFINED);
+    dataset.write().mode(SaveMode.Overwrite).saveAsTable(FAILING_TABLE_NAME_FAIL_ON_APPLY);
+    // creating table that way should fail because of a failing visitor defined in
+    // TestOpenLineageEventHandlerFactory
+
+    StaticExecutionContextFactory.waitForExecutionEnd();
+    assertEquals(1, spark.table(FAILING_TABLE_NAME_FAIL_ON_IS_DEFINED).count());
+    assertEquals(1, spark.table(FAILING_TABLE_NAME_FAIL_ON_APPLY).count());
```
Comment (collado-mike) [inline integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java:127]: Can we add one more assertion on the message sent to the backend (if any)?

**Review by collado-mike — APPROVED** [review]
These are great! I think it would be good to add one more assertion, but then feel free to merge.

File: integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java (hunk 2 above)
Comment (pawel-big-lebowski) [reply integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java:131]: ok, good point.

File: integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java (hunk 1 above)
Comment (pawel-big-lebowski) [reply integration/spark/src/test/common/java/io/openlineage/spark/agent/SparkOpenLineageFailuresTest.java:92]: thanks!

**Pull request merged**